### ISSUE TITLE
Dialog replication question

### ISSUE THREAD

**Issue by kertor** [OpenSIPS/opensips#1597]
Hello guys! Can you please recommend how best to set up dialog replication in this scheme:
```
   EXT-IP                              Internal-Network   
     |                                      |       |<--------->(10.0.0.3)opensips-node-1<--|
     |(1.1.1.1)opensips-1-balancer(10.0.0.1)|<----->|                                       |
     |                                      |       |                                       |Clusterer with
     |                                      |       |                                       |dialog replication
     |(1.1.1.1)opensips-2-balancer(10.0.0.2)|<----->|                                       |
     |                                      |       |<--------->(10.0.0.4)opensips-node-2<--|
     |                                      |
```
Balancers: I have routing with OSPF in EXT-IP side. "opensips-1-balancer" and"opensips-2-balancer" have the same external IP address, but these balancers do not have 'dialog' module.
Balancers routing requests to nodes using the module 'dispatcher'.

Nodes: Here I use module 'dialog' and 'clusterer'. No VirtualIP or Anycast IP on interfaces. Is anyway to configure dialog replication between nodes withoud using VIP or Anycast?
In my scheme "Nodes" is main element of the system. And i need configure failover here. In logs i see this errors:
```
Jan 28 12:37:17 [50365] ERROR:dialog:receive_dlg_repl: Failed to process a binary packet!
Jan 28 12:37:17 [50365] WARNING:dialog:fetch_socket_info: non-local socket <udp:10.0.0.4:5060>...ignoring
Jan 28 12:37:17 [50365] WARNING:dialog:fetch_socket_info: non-local socket <udp:10.0.0.4:5060>...ignoring
Jan 28 12:37:17 [50365] ERROR:dialog:receive_dlg_repl: Failed to process a binary packet!
Jan 28 12:37:17 [50365] WARNING:dialog:fetch_socket_info: non-local socket <udp:10.0.0.4:5060>...ignoring
Jan 28 12:37:17 [50365] WARNING:dialog:fetch_socket_info: non-local socket <udp:10.0.0.4:5060>...ignoring
Jan 28 12:37:17 [50365] ERROR:dialog:receive_dlg_repl: Failed to process a binary packet!
Jan 28 12:37:17 [50365] WARNING:dialog:fetch_socket_info: non-local socket <udp:10.0.0.4:5060>...ignoring
Jan 28 12:37:17 [50365] WARNING:dialog:fetch_socket_info: non-local socket <udp:10.0.0.4:5060>...ignoring
Jan 28 12:37:17 [50365] ERROR:dialog:receive_dlg_repl: Failed to process a binary packet!
```

Here http://opensips-open-sip-server.1449251.n2.nabble.com/dialog-replication-td7611780.html i see Liviu comment:
```
Dialog replication is _built_ to work with VIPs.  If you are not using 
VIPs, you will only be able to share dialog profiles [1] between your 
public front-end boxes. 
```

Is anyway configure dialog replication in my scheme without using VIP or Anycast? Thanks!

**Comment by nikbyte** [comment]
Bogdan, LBs are stateless, they distribute calls via dispatcher by hash.

**Comment by nikbyte** [comment]
Both LBs are at the same IPs, they’re stateless and doesn’t matter from what TS they receive call. LBs have no dialog module inside, only transaction level (TM module).

**Comment by bogdan-iancu** [comment]
So you say the LB (which sandwiches the OpenSIPS with dialog state) does not route in dialog requests based on IP (record route or contact), but via dispatched with callid hashing (alg 0 - to ensure all requests with the same callid to reach the same server) ? or ?

**Comment by nikbyte** [comment]
yes. exactly. it uses hash by callid.

**Comment by bogdan-iancu** [comment]
an short update here - this need was addressed by adding tagging support for the listening interfaces. The tagging offers a way to virtually correlate interfaces on different opensips / servers. Like if a dialog was created via the interface/listener A (on server A), it will have a corresponding interface (during replication) on server B is there is an interface/listener B with the same tag as A. 
Something like:
On server A:
listen=udp:10.0.0.15:5060 tag my_priv_interface

On server B:
listen=udp:10.0.0.17:5060 tag my_priv_interface

This support was added in trunk via commits e6881b621a6dde845b57b623829c728025798277..5a56d90a991482fa635679611bf228312b751853 .

**Comment by bogdan-iancu** [comment]
Hey @petekelly - if so, than it is a bug, not a feature :P . Are those OPTIONS originated by nathelper, dispatcher, or ??

**Comment by petekelly** [comment]
@bogdan-iancu it’s the drouting module!

**Comment by bogdan-iancu** [comment]
Do you force in drouting the probing socket via per-module or per-gateway setting ?

**Comment by petekelly** [comment]
It’s per gateway - and am also seeing the bug in the invite too.

**Comment by bogdan-iancu** [comment]
thanks @petekelly , I traced back the problem to commit d5f2f69fac093282dba697bdbb43adc3f3549624 - it seems the new code fails to properly aggregate *all* the parameters of the listener.
@razvancrainea is working on a fix

**Comment by razvancrainea** [comment]
Hey! I've just pushed a fix for this in 939213d666d7, thanks for reporting this!

**Comment by petekelly** [comment]
Thanks Razvan, I have just tested this and it looks like the fix has worked.

Much appreciated!

Pete
